substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JmxServerFeature.java

```diff
@@ -108,5 +119,8 @@ private static void configureReflection() {
             Class<?> clazz = p.getClass();
             RuntimeReflection.register(clazz);
         }
+
+        RuntimeReflection.register(access.findClassByName("com.sun.jmx.remote.protocol.rmi.ServerProvider"));
```

Comment:
I'd be nice if we could add a reason for each registration, similar to how `RuntimeClassInitializationSupport#register()` accepts a reason (see for example `JDKInitializationFeature`). But I don't think we have anything for this yet, right @cstancu?